src/js/tools/sentryMetroSerializer.ts

```diff
@@ -0,0 +1,256 @@
+  const match = code.match(
+    /sentry-dbid-([0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12})/,
+  );
+
+  if (match) {
+    return match[1];
+  } else {
+    return undefined;
+  }
+}
+
+/**
+ * Creates a Metro serializer that adds Debug ID module to the plain bundle.
+ * The Debug ID module is a virtual module that provides a debug ID in runtime.
+ *
+ * RAM Bundles do not support custom serializers.
+ */
+export const createSentryMetroSerializer = (customSerializer?: MetroSerializer): MetroSerializer => {
+  return async function (this: ExpectedSerializedConfigThisContext, entryPoint, preModules, graph, options) {
+    let currentDebugIdModule: Module<MixedOutput> | undefined;
+    const serializer = customSerializer || createDefaultMetroSerializer(this);
+    options.sentryBundleCallback = sentryBundleCallback;
+
+    // Add debug id module to the preModules
+    let modifiedPreModules: readonly Module<MixedOutput>[];
+    const containsDebugIdModule = preModules.some(module => module.path === DEBUG_ID_MODULE_PATH);
+    if (!containsDebugIdModule) {
+      const debugIdModule = createDebugIdModule(DEBUG_ID_PLACE_HOLDER);
+      const tmpPreModules = [...preModules];
+      if (tmpPreModules.length > 0 && tmpPreModules[0] !== undefined && tmpPreModules[0].path === PRELUDE_MODULE_PATH) {
+        // prelude module must be first as it measures the bundle startup time
+        tmpPreModules.unshift(preModules[0]);
+        tmpPreModules[1] = debugIdModule;
+      } else {
+        tmpPreModules.unshift(debugIdModule);
+      }
+      modifiedPreModules = tmpPreModules;
+    } else {
+      modifiedPreModules = preModules;
+    }
+
+    let bundleCode: string = '';
+    let bundleMapString: string = '{}';
+
+    // Run wrapped serializer
+    const serializerResult = serializer(entryPoint, modifiedPreModules, graph, options);
+    if (typeof serializerResult === 'string') {
+      bundleCode = serializerResult;
+    } else if ('map' in serializerResult) {
+      bundleCode = serializerResult.code;
+      bundleMapString = serializerResult.map;
+    } else {
+      const awaitedResult = await serializerResult;
+      if (typeof awaitedResult === 'string') {
+        bundleCode = awaitedResult;
+      } else {
+        bundleCode = awaitedResult.code;
+        bundleMapString = awaitedResult.map;
+      }
+    }
+
+    // Add debug id comment to the bundle
+    const debugId = determineDebugIdFromBundleSource(bundleCode);
+    if (!debugId) {
+      throw new Error(
+        'Debug ID was not found in the bundle. Call `options.sentryBundleCallback` if you are using a custom serializer.',
+      );
+    }
+    if (graph.transformOptions.hot !== true) {
+      // eslint-disable-next-line no-console
+      console.log('info ' + `Bundle Debug ID: ${debugId}`);
+    }
+
+    currentDebugIdModule?.output[0].data.code.replace(DEBUG_ID_PLACE_HOLDER, debugId);
```

Comment:
l: if `currentDebugIdModule` is not defined, should we just early return above?

Reply:
There is already an early return when the Debug ID was not found in the final bundle code. 

I've adjusted the code so there is always Debug ID module either existing one or created by us.